consensus/beacon/consensus.go

```diff
@@ -461,6 +476,15 @@ func (beacon *Beacon) SetThreads(threads int) {
 // It depends on the parentHash already being stored in the database.
 // If the parentHash is not stored in the database a UnknownAncestor error is returned.
 func IsTTDReached(chain consensus.ChainHeaderReader, parentHash common.Hash, parentNumber uint64) (bool, error) {
+	if cfg := chain.Config(); cfg.Optimism != nil {
+		num := parentNumber
+		if num == ^(uint64(0)) { // caller can (intentionally?!) underflow on parent-of-block 0 case.
```

Comment:
When does this occur? When checking if IsTTDReached for the genesis block?

Reply:
Yes exactly